tensorflow/core/kernels/non_max_suppression_op.cc

```diff
@@ -522,6 +533,8 @@ class NonMaxSuppressionOp : public OpKernel {
     int num_boxes = 0;
     ParseAndCheckBoxSizes(context, boxes, &num_boxes);
     CheckScoreSizes(context, num_boxes, scores);
+    // check if the boxes have logical coordinates
+    OP_REQUIRES(context, CheckBoxesCoordinates<float>(boxes, num_boxes), errors::InvalidArgument("boxes coordinates shouldn't have x1=x2 or y1=y2"));
```

Comment:
I would rewrite this to one of:

- `Boxes are empty (x1 == x2 or y1 == y2)`
- `Boxes have a dimension of size 0 (x1 == x2 or y1 == y2)`

Or the same options but without the condition in the paranthesis.

Reply:
Thank you for your advice. I corrected those mistakes.